multicluster/cmd/multicluster-controller/gateway_webhook.go

```diff
@@ -59,6 +61,25 @@ func (v *gatewayValidator) Handle(ctx context.Context, req admission.Request) ad
 		klog.ErrorS(err, "failed to create Gateway", "Gateway", klog.KObj(gateway), "Namespace", v.namespace)
 		return admission.Errored(http.StatusPreconditionFailed, err)
 	}
+	if req.Operation == admissionv1.Update {
+		oldGateway := &mcv1alpha1.Gateway{}
+		if req.OldObject.Raw != nil {
+			if err := json.Unmarshal(req.OldObject.Raw, oldGateway); err != nil {
+				klog.ErrorS(err, "Error while decoding old Gateway")
+				return admission.Errored(http.StatusBadRequest, err)
+			}
+		}
+		ui := req.UserInfo
+		_, saName, err := serviceaccount.SplitUsername(ui.Username)
+		if err != nil {
+			klog.ErrorS(err, "Error getting ServiceAccount name", "Gateway", req.Namespace+"/"+req.Name)
+			return admission.Errored(http.StatusBadRequest, err)
+		}
+		if saName != mcControllerSAName {
+			return admission.Errored(http.StatusPreconditionFailed, fmt.Errorf("ServiceCIDR can only be updated by Antrea Multi-cluster Controller"))
```

Comment:
You are not comparing `ServiceCIDR`, so better to say `Gateway can only be updated by Antrea Multi-cluster Controller`

Reply:
Fixed. Removed the ServiceAccount to the above of the Operation check. Now only SA is "antrea-mc-controller" could create or update Gateway.